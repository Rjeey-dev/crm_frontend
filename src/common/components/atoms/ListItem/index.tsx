import React, {FormEvent, ReactNode} from "react";

interface IProps {
    children: ReactNode,
    classes?: string,
    handleClick?: (event: FormEvent<HTMLLIElement>) => void,
    attributes?: any
}

function ListItem(props: IProps) {
    return <li className={props.classes} onClick={props.handleClick} {...props.attributes}>{props.children}</li>
}

export default ListItem;