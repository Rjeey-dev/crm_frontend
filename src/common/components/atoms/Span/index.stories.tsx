import React from 'react';
import { storiesOf } from '@storybook/react';
import Span from './index';

storiesOf('Span', module)
    .add('With text', () => (
        <Span classes='test' text='Some text'/>
    ))
    .add('Without text', () => (
        <Span classes='test'/>
    ))
    .add('Without classes', () => (
        <Span text='Some text'/>
    ))
    .add('Without classes and text', () => (
        <Span/>
    ));