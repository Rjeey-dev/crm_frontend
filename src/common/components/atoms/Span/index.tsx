import React, {ReactNode, ReactNodeArray} from "react";

interface IProps {
    classes?: string,
    id?: string,
    children?: ReactNodeArray | ReactNode,
    attributes?: object,
    onClick?: (event: React.FormEvent<HTMLSpanElement>) => void
}

function Span(props: IProps) {
    return <span onClick={props.onClick} className={props.classes} {...props.attributes} id={props.id}>{props.children}</span>;
}

export default Span;
