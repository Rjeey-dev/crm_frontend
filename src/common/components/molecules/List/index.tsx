import React, {ReactNode} from "react";

interface IProps {
    children: ReactNode,
    classes?: string,
    id?: string
}

export default function List(props: IProps) {
    return <ul className={props.classes} id={props.id}>
        {props.children}
    </ul>
}
