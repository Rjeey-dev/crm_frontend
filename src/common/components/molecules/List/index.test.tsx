import {shallow} from "enzyme";
import React from 'react';

import ListItem from "../../atoms/ListItem/index";
import Span from "../../atoms/Span/index";
import List from './index';

it("Renders <List/> with correct properties", () => {
    const output = shallow(
        <List classes='class'>
            <ListItem>
                <Span/>
            </ListItem>
        </List>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <List/> with only required props", () => {
    const output = shallow(
        <List>
            <ListItem>
                <Span/>
            </ListItem>
        </List>
    ).html();

    expect(output).toMatchSnapshot();
});