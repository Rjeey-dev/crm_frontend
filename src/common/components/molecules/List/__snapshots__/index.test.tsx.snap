// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <List/> with correct properties 1`] = `"<ul class=\\"class\\"><li><span></span></li></ul>"`;

exports[`Renders <List/> with only required props 1`] = `"<ul><li><span></span></li></ul>"`;
